src/mlpack/methods/bayesian_ridge/bayesian_ridge.hpp

```diff
@@ -0,0 +1,298 @@
+/**
+ * @file bayesian_ridge.hpp
+ * @author Clement Mercier
+ *
+ * Definition of the BayesianRidge class, which performs the
+ * bayesian linear regression. According to the armadillo standards,
+ * all the functions consider data in column-major format.
+**/
+#ifndef MLPACK_METHODS_BAYESIAN_RIDGE_BAYESIAN_RIDGE_HPP
+#define  MLPACK_METHODS_BAYESIAN_RIDGE_BAYESIAN_RIDGE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack{
+namespace regression{
+/**
+ * This class implements the bayesian linear regression. "Bayesian treatment
+ * of linear regression, which will avoid the over-fitting problem of maximum
+ * likelihood, and which will also lead to automatic methods of determining
+ * model complexity using the training data alone.", C.Bishop.
+ *
+ * More details and description in :
+ * Christopher Bishop (2006), Pattern Recognition and Machine Learning.
+ * David J.C MacKay (1991), Bayesian Interpolation, Computation and Neural
+ * systems.
+ 
+ * Model optimization is automatic and does not require cross validation
+ * procedure to be optimized.
+ *
+ * @code
+ * @article{MacKay91bayesianinterpolation,
+ *   author = {David J.C. MacKay},
+ *   title = {Bayesian Interpolation},
+ *   journal = {NEURAL COMPUTATION},
+ *   year = {1991},
+ *   volume = {4},
+ *   pages = {415--447}
+ * }
+ * @endcode
+ *
+ * @book{Bishop:2006:PRM:1162264,
+ *   author = {Bishop, Christopher M.},
+ *   title = {Pattern Recognition and Machine Learning (Information Science 
+ *            and Statistics)},
+ *   chapter = {3}
+ *   year = {2006},
+ *   isbn = {0387310738},
+ *   publisher = {Springer-Verlag},
+ *   address = {Berlin, Heidelberg},
+ * } 
+ * @encode
+ *   
+ * Example of use:
+ *
+ * @code
+ * arma::mat Xtrain; // Train data matrix. Column-major.
+ * arma::rowvec ytrain; // Train target values.
+ 
+ * // Train the model. Regularization strength is optimally tunned with the
+ * // training data alone by applying the Train method.
```

Comment:
Do you mean here that there is no need to specify any regularization, since that is implicit in the Bayesian approach?

Reply:
Exactly, I add this precision in the description.

Reply:
Thanks, I really like the paragraph you added about the advantage of this vs. linear or ridge regression.